qa/tasks/salt.py

```diff
@@ -0,0 +1,203 @@
+            sh('scp -3 {}:salt/minion-keys/{}.* {}:'.format(
+                self.sm.master_remote.name,
+                minion_id, rem.name))
+            sudo_write_file(rem, '/etc/salt/minion_id', minion_id)
+            #
+            # set proper owner and permissions on keys
+            rem.run(
+                args=[
+                    'sudo',
+                    'chown',
+                    'root',
+                    '{}.pem'.format(minion_id),
+                    '{}.pub'.format(minion_id),
+                    run.Raw(';'),
+                    'sudo',
+                    'chmod',
+                    '600',
+                    '{}.pem'.format(minion_id),
+                    run.Raw(';'),
+                    'sudo',
+                    'chmod',
+                    '644',
+                    '{}.pub'.format(minion_id),
+                ],
+            )
+            #
+            # move keys to correct location
+            move_file(rem, '{}.pem'.format(minion_id),
+                      '/etc/salt/pki/minion/minion.pem', sudo=True,
+                      preserve_perms=False)
+            move_file(rem, '{}.pub'.format(minion_id),
+                      '/etc/salt/pki/minion/minion.pub', sudo=True,
+                      preserve_perms=False)
+
+    def __set_minion_master(self):
+        """Points all minions to the master"""
+        master_id = self.sm.master_remote.hostname
+        for rem in self.remotes.iterkeys():
+            # remove old master public key if present. Minion will refuse to
+            # start if master name changed but old key is present
+            delete_file(rem, '/etc/salt/pki/minion/minion_master.pub',
+                        sudo=True, check=False)
+
+            # set master id
+            sed_cmd = ('echo master: {} > '
+                       '/etc/salt/minion.d/master.conf').format(master_id)
+            rem.run(args=[
+                'sudo',
+                'sh',
+                '-c',
+                sed_cmd,
+            ])
+
+    def __set_debug_log_level(self):
+        """Sets log_level: debug for all salt daemons"""
+        for rem in self.remotes.iterkeys():
+            rem.run(args=[
+                'sudo',
+                'sed', '--in-place', '--regexp-extended',
+                '-e', 's/^\s*#\s*log_level:.*$/log_level: debug/g',
+                '-e', '/^\s*#.*$/d', '-e', '/^\s*$/d',
+                '/etc/salt/master',
+                '/etc/salt/minion',
+            ])
+
+    def __provision_minions(self):
+        self.__generate_minion_keys()
+        self.__preseed_minions()
+        self.__set_minion_master()
+        self.__set_debug_log_level()
+        self.sm.start_master()
+
+    def setup(self):
+        super(Salt, self).setup()
+        log.info("Salt task setup...")
+        self.__provision_minions()
+
+    def begin(self):
+        log.info("Salt task begin...")
```

Comment:
log.debug("beginning of Salt task begin method")
...
log.debug("end of Salt task begin method")